### PR TITLE
GH-296: Arrow keys work with `masq`

### PR DESCRIPTION
**Pull request by dnwiebe**

### CODE REVIEW

File: masq/src/line_reader.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+impl BufRead for LineReader {
+    fn fill_buf(&mut self) -> Result<&[u8], io::Error> {
+        panic!("Should never be called");
+    }
+
+    fn consume(&mut self, _: usize) {
+        panic!("Should never be called");
+    }
+
+    fn read_line(&mut self, buf: &mut String) -> Result<usize, io::Error> {
```
Comment (dnwiebe) [inline masq/src/line_reader.rs:27]: Tests missing for this method

File: masq/src/line_reader.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        let line = match self.delegate.readline(MASQ_PROMPT) {
+            Ok(line) => line,
+            Err(e) => unimplemented!("{:?}", e),
+        };
```
Comment (bertllll) [inline masq/src/line_reader.rs:41]: Replace this unimplemented with something more suitable.

File: masq/src/main.rs [hunk 3]
```diff
@@ -183,9 +184,41 @@ mod tests {
...
+    struct BufReadFactoryMock {
+        interactive: RefCell<Option<ByteArrayReader>>,
+    }
+
+    impl BufReadFactory for BufReadFactoryMock {
+        fn make_interactive(&self) -> Box<dyn BufRead> {
+            Box::new(self.interactive.borrow_mut().take().unwrap())
+        }
+
+        fn make_non_interactive(&self) -> Box<dyn BufRead> {
```
Comment (bertllll) [inline masq/src/main.rs]: Remove this method from the trait.

File: masq/src/line_reader.rs (hunk 2 above)
Comment (bertllll) [reply masq/src/line_reader.rs:41]: Great. Resolved with elegance.

**Review by bertllll — APPROVED** [review]

**Pull request merged**